Dockerfile

```diff
@@ -1,16 +1,19 @@
-FROM debian:jessie
+FROM debian:buster-slim
+RUN apt-get update && apt-get dist-upgrade -y && \
+    apt-get install -y ca-certificates git curl jq build-essential libluajit-5.1-2 luarocks python3-setuptools python3-dev python3-pip
+RUN luarocks install lua-cjson
 
-RUN apt-get update -y -qq && apt-get install -y ca-certificates curl wget build-essential libluajit-5.1-2 libluajit-5.1-dev luarocks python3-setuptools python3-dev libpython3.4 locales ruby-dev gcc make rubygems rpm gnupg expect perl git openssh-client procps autoconf autogen libtool
+RUN pip3 install grpcio
+RUN pip3 install protobuf
 
-RUN wget -O go.tgz "https://dl.google.com/go/go1.10.3.linux-amd64.tar.gz" && tar -C /usr/local -xzf go.tgz && rm go.tgz
-RUN mkdir -p "/go/src" "/go/bin" && chmod -R 777 "/go"
+# Go install
+RUN curl -sL https://dl.google.com/go/go1.12.8.linux-amd64.tar.gz | tar -xzC /usr/local/
```

Comment:
is there a reason we don't use `FROM golang:1.12.8-buster`? Rather than installing go into `debian:buster-slim`?

Also, do we still need to install things like `lua`, `python`, `protobuf`?

Reply:
You make a fair point. However, I do start from buster-slim which is smaller than the regular buster. A small difference in size builds up over (hopefully millions, if not billions) of pulls.

Reply:
lua python and protobuf are requirements for this env. We are not seeking for minimal image here, it can be any size. This is not a kind of image which is used on day to day basis, like gw or dash images, it used very rarely to build things.

Reply:
Also I suggest going with simple way first and improving it later.